go/test/endtoend/vtgate/queries/random/random_expr_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package random
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/slices2"
+	"vitess.io/vitess/go/vt/sqlparser"
+)
+
+func TestRandomExprWithTables(t *testing.T) {
+	schemaTables := []tableT{
+		{name: sqlparser.NewTableName("emp")},
+		{name: sqlparser.NewTableName("dept")},
+	}
+	schemaTables[0].addColumns([]column{
+		{name: "empno", typ: "bigint"},
+		{name: "ename", typ: "varchar"},
+		{name: "job", typ: "varchar"},
+		{name: "mgr", typ: "bigint"},
+		{name: "hiredate", typ: "date"},
+		{name: "sal", typ: "bigint"},
+		{name: "comm", typ: "bigint"},
+		{name: "deptno", typ: "bigint"},
+	}...)
+	schemaTables[1].addColumns([]column{
+		{name: "deptno", typ: "bigint"},
+		{name: "dname", typ: "varchar"},
+		{name: "loc", typ: "varchar"},
+	}...)
+
+	seed := time.Now().UnixNano()
+	g := sqlparser.NewGenerator(seed, 3, slices2.Map(schemaTables, func(t tableT) sqlparser.ExprGenerator { return &t })...)
+	randomExpr, _ := g.Expression()
+	fmt.Println(sqlparser.String(randomExpr))
```

Comment:
no need to print here. you could do something like: 

```go
_, _ = g.Expression()
```

and then leave it at that. this tests that schematables implements the correct interface and that nothing panics when using the random generator